test/compiler.test.js

```diff
@@ -1564,6 +1564,21 @@ describe('compiler', function() {
         .to.throw(/path-does-not-exist/);
     });
 
+    it(
+      'does not fail when an optional middleware cannot be resolved',
+      function() {
```

Comment:
Please keep the formatting consistent with the code around, see e.g. the test below.

```js
it('does not fail...',
  function() {
    // ...
  });
```